File: bus_buddy_user/lib/features/model/stop.dart
import 'package:bus_buddy_user/features/model/place.dart';
import 'package:bus_buddy_user/features/model/route.dart';

class Stop {
  int? id;
  int? stopTime;
  int? orderNo;
  Place? place;
  RouteModel? route;
  String? createdAt;
  String? updatedAt;

  Stop(
      {this.id,
      this.stopTime,
      this.orderNo,
      this.place,
      this.route,
      this.createdAt,
      this.updatedAt});

  Stop.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    stopTime = json['stopTime'];
    orderNo = json['orderNo'];
    place = json['place'] != null ? Place.fromJson(json['place']) : null;
    route = json['route'] != null ? RouteModel.fromJson(json['route']) : null;
    createdAt = json['createdAt'];
    updatedAt = json['updatedAt'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['stopTime'] = stopTime;
    data['orderNo'] = orderNo;
    if (place != null) {
      data['place'] = place!.toJson();
    }
    if (route != null) {
      data['route'] = route!.toJson();
    }
    data['createdAt'] = createdAt;
    data['updatedAt'] = updatedAt;
    return data;
  }
}

File: bus_buddy/src/main/java/com/np/suprimpoudel/bus_buddy/entities/Stop.java
package com.np.suprimpoudel.bus_buddy.entities;

import jakarta.persistence.*;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.util.Date;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Stop {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;

    @Min(value = 1, message = "Stop time must be at least 1 minute")
    @Column(name = "stop_time", nullable = false)
    private int stopTime;

    @Min(value = 1, message = "Invalid order number")
    @Column(name = "order_no", nullable = false)
    private int orderNo;

    @ManyToOne
    @NotNull(message = "Place is required")
    @JoinColumn(name = "place", referencedColumnName = "id")
    private Place place;

    @ManyToOne
    @NotNull(message = "Route is required")
    @JoinColumn(name = "route", referencedColumnName = "id")
    private Route route;

    @Temporal(TemporalType.TIMESTAMP)
    @CreationTimestamp
    @Column(name = "created_at")
    private Date createdAt;

    @Temporal(TemporalType.TIMESTAMP)
    @UpdateTimestamp
    @Column(name = "updated_at")
    private Date updatedAt;
}
